chaos_genius/controllers/data_source_metadata_controller.py

```diff
@@ -0,0 +1,227 @@
+from chaos_genius.controllers.data_source_controller import get_datasource_data_from_id
+from chaos_genius.connectors import (
+    get_sqla_db_conn,
+    get_schema_names,
+    get_table_list,
+    get_table_info,
+)
+
+from chaos_genius.utils.metadata_api_config import TABLE_VIEW_MATERIALIZED_VIEW_AVAILABILITY
+
+NON_THIRD_PARTY_DATASOURCES = TABLE_VIEW_MATERIALIZED_VIEW_AVAILABILITY.keys()
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_schema_list(data_source, as_obj=False):
+    """Fetch the schema list from the metadata of the given data source."""
+    schema_list = []
+    data_source_metadata = (
+        DataSourceMetadata.query.filter(
+            DataSourceMetadata.data_source_id == data_source,
+            DataSourceMetadata.metadata_type == "schema_list",
+        )
+        .order_by(DataSourceMetadata.created_at.desc())
+        .first()
+    )
+    if data_source_metadata:
+        schema_list = data_source_metadata.metadata_info.get("schema_list", [])
+
+    if as_obj:
+        return data_source_metadata
+    else:
+        return schema_list
+
+
+def fetch_table_list(data_source, schema, as_obj=False):
+    """Fetch the table list from the metadata of the given data source and schema."""
+    table_list = []
+    data_source_metadata = (
+        DataSourceMetadata.query.filter(
+            DataSourceMetadata.data_source_id == data_source,
+            DataSourceMetadata.metadata_type == "table_list",
+            DataSourceMetadata.metadata_param == get_metadata_param_str([schema]),
+        )
+        .order_by(DataSourceMetadata.created_at.desc())
+        .first()
+    )
+    if data_source_metadata:
+        table_list = data_source_metadata.metadata_info.get("table_list", [])
+
+    if as_obj:
+        return data_source_metadata
+    else:
+        return table_list
+
+
+def delete_table_list(data_source, schema):
+    """delete the table list from the metadata of the given data source and schema."""
+    data_source_metadata = (
+        DataSourceMetadata.query.filter(
+            DataSourceMetadata.data_source_id == data_source,
+            DataSourceMetadata.metadata_type == "table_list",
+            DataSourceMetadata.metadata_param == get_metadata_param_str([schema]),
+        )
+        .order_by(DataSourceMetadata.created_at.desc())
+        .first()
+    )
+    if data_source_metadata:
+        data_source_metadata.delete(commit=True)
+
+
+def fetch_table_info(data_source, schema, table, as_obj=False):
+    """Fetch the table info from the metadata of the given data source and table"""
+    table_info = {}
+    data_source_metadata = (
+        DataSourceMetadata.query.filter(
+            DataSourceMetadata.data_source_id == data_source,
+            DataSourceMetadata.metadata_type == "table_info",
+            DataSourceMetadata.metadata_param
+            == get_metadata_param_str([schema, table]),
+        )
```

Comment:
am not 100% sure about this as it would be minor improvement as abstracted func would also take same parameters and the function call will be more or less similar with some readability improvement just bcoz we remove .first .orderby .query.filter and lengthy param names to filter. I see that it gives some improvement but not 100% convinced